music21/musicxml/m21ToXml.py

```diff
@@ -2321,8 +2326,8 @@ def metadataToMiscellaneous(self, md=None):
         >>> mxMisc = SX.metadataToMiscellaneous(md)
         >>> SX.dump(mxMisc)
         <miscellaneous>
-          <miscellaneous-field name="date">1689/--/-- or earlier</miscellaneous-field>
-          <miscellaneous-field name="localeOfComposition">Rome</miscellaneous-field>
+          <miscellaneous-field name="dcterms:created">1689/--/-- or earlier</miscellaneous-field>
+          <miscellaneous-field name="humdrum:OPC">Rome</miscellaneous-field>
```

Comment:
I think I prefer localeOfComposition here -- or in any case not using humdrum TLA (three-letter-acronyms)

Reply:
In the miscellaneous tag, I think we are dealing with "how we exchange metadata with others", and so I thought that 'namespace:name' was better (otherwise we would have to publish all music21's names somewhere, and we've made a _new_ namespace).  I'm not real fond of the TLAs either, but...

Reply:
okay -- it's not a deal breaker.

Reply:
I'm definitely okay with dcterms:... codes there, because they're relatively well accepted.  But the marcrel terms aren't presently parseable by anything else but this, and they'll show up in notation software's miscellaneous metadata editors as something incomprehensible rather than as something human readable.  I guess, when I put into music21 "localeOfComposition = ..." or "transcriber = ..." I'd prefer to see it in Finale as "localeOfComposition=..." and "transcriber=..." than "humdrum:OPC=...." and "marcrel:TRC=...".  We could make a keyword argument in `.write(useNamespacedMetadata=True)` to get the current output, but it shouldn't be the default.

Reply:
If you **don't** write the namespace you will probably have a hard time round-tripping the data, since you will be stripping the "source" of the code. `marcrel:trc` or `humdrum:OPC` will become just `trc` or `OPC`, which would probably be even worse, since there would be almost no hope of being able to figure out the original vocabulary from which the code was derived, and there would be no way of 'reattaching' it back to the right namespace without a big lookup table containing every possible value and it's associated namespace (tricky if there are two of the same values from different namespaces....)

If you wanted to change the value of `name` to something more human-readable like `name="Locale of Composition"` or something then yes, it's human-readable, but also music21-specific which makes interchange harder since you haven't tied in to an existing spec. You would also probably want an import converter that changed the 'human readable' label to a 'machine-actionable' label if you were to open a file to make processing easier, but that might get tricky.

(It's too bad MusicXML only accepts the `name` attribute for this tag. In MEI we would attach `label`, `auth`, and `auth.uri` attributes to the same element, making it possible to capture both the human and machine codes at the same time.)

On the subject of 'familiarity', don't forget that the prefixes are just aliases to the full URI -- `marcrel:trc` is short for `http://id.loc.gov/vocabulary/relators/trc`, which is the actual fully-qualified value, and the thing that gives it universal uniqueness in the global space. There is no magic in the `marcrel` prefix, and nothing preventing you from changing `marcrel` to anything you want -- `relators:trc` or `relationships:trc` are all possible as long as you keep them unique (the same prefix can't map to two different URIs in your document), and retain the mapping of prefix -> URI stored somewhere accessible within your software.

Reply:
music21 can roundtrip "localOfComposition" -- but of course it's music21 specific; what I haven't determined is if the marcrel prefix plus TLA is used enough to make it worth sacrificing readability for the sake of (hypothetical?) roundtrips to another parsing system.  Whenever I've googled anything I'm finding 4,000 hits, maybe 20,000 hits, but nothing like the 5-10 million hits for Dublin Core, so I'm not convinced yet that Marcrel or MARC Relators, etc. has the sort of audience that I'm willing to commit my time to maintaining connections to for the next decade.  There's a bit of scope-creep is all.  It was a tough choice for me to decide, "yes, Dublin Core is worth it."  MARC wasn't anywhere in the initial proposal.  Thanks!

Reply:
Yeah, I get that. Unfortunately, MARC relator codes are actually one of the few "standards" for a common vocabulary relating a person to a document, and are probably (by far) the most used. As I think I mentioned earlier, the MARC relator codes are the ones endorsed by Dublin Core itself.

Other standard possibilities include the Getty vocabularies, but they're even more esoteric and harder to figure out how to apply. Or the RDA Vocabularies but .... well, here's an example: http://www.rdaregistry.info/Elements/w/#P10053

Not something I would necessarily foist on an unsuspecting public. :-D

Googling "MARC Relator Codes" gives me about 540,000 hits. (They were invented far before prefixes were 'normal' so using 'marcrel' or some short form isn't likely to find much.)

As I mentioned, if you've got the full URI for the codes stored somewhere, then the likelihood that you're not able to find a reference to them is low in the future; also, it's highly unlikely that they will change anytime soon -- that would necessitate a vast amount of work in library catalogue systems!